qcog_python_client/qcog/qcogasync.py

```diff
@@ -0,0 +1,308 @@
+"""Async wrapper."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import pandas as pd
+
+from qcog_python_client.qcog._baseclient import BaseQcogClient
+from qcog_python_client.qcog._data_uploader import DataClient
+from qcog_python_client.qcog._httpclient import RequestClient
+from qcog_python_client.qcog._interfaces import (
+    ABCDataClient,
+    ABCRequestClient,
+)
+from qcog_python_client.qcog._version import DEFAULT_QCOG_VERSION
+from qcog_python_client.schema.common import (
+    InferenceParameters,
+    Matrix,
+    NotRequiredStateParams,
+    NotRequiredWeightParams,
+)
+from qcog_python_client.schema.generated_schema.models import TrainingStatus
+
+
+class AsyncQcogClient(BaseQcogClient):
+    """Async Qcog Client."""
+
+    @classmethod
+    async def create(
+        cls,
+        *,
+        token: str | None = None,
+        hostname: str = "dev.qognitive.io",
+        port: int = 443,
+        api_version: str = "v1",
+        safe_mode: bool = False,
+        version: str = DEFAULT_QCOG_VERSION,
+        httpclient: ABCRequestClient | None = None,
+        dataclient: ABCDataClient | None = None,
+    ) -> AsyncQcogClient:
+        """Create a new Qcog client.
+
+        TODO: docstring
+        """
+        client = cls()
```

Comment:
it would be nice to make a call to `BaseQcogClient._create` in the body of this method to re-use same code stored in the base class

Reply:
I'm not 100% sure of what's the behavior of `classmethods` within inheritance.
If I'm calling `BaseQcogClient._create` I'm getting back the parent class, but I want to instantiate the Child class

Reply:
ah, good point!